### PR TITLE
Convert ProfileService to suspend functions

### PR DESCRIPTION
**Pull request by Dominaezzz**
Signed-off-by: Dominic Fischer <dominicfischer7@gmail.com>

### Pull Request Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [x] Pull request is based on the develop branch
- [ ] Pull request updates [CHANGES.md](https://github.com/vector-im/element-android/blob/develop/CHANGES.md)
- [x] Pull request includes screenshots or videos if containing UI changes
- [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/profile/ProfileService.kt [hunk 1]
```diff
@@ -97,28 +95,26 @@ interface ProfileService {
...
+    suspend fun submitSmsCode(threePid: ThreePid.Msisdn, code: String)
 
     /**
      * Finalize adding a 3Pids. Call this method once the user has validated that he owns the ThreePid
      */
-    fun finalizeAddingThreePid(threePid: ThreePid,
-                               userInteractiveAuthInterceptor: UserInteractiveAuthInterceptor,
-                               matrixCallback: MatrixCallback<Unit>): Cancelable
+    suspend fun finalizeAddingThreePid(threePid: ThreePid,
+                               userInteractiveAuthInterceptor: UserInteractiveAuthInterceptor)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/profile/ProfileService.kt:109]: (really minor) formatting issue

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/DefaultProfileService.kt [hunk 2]
```diff
@@ -55,64 +53,38 @@ internal class DefaultProfileService @Inject constructor(private val taskExecuto
...
-
-                        override fun onFailure(failure: Throwable) {
-                            matrixCallback.onFailure(failure)
-                        }
-                    }
-                }
-                .executeBy(taskExecutor)
+        val data = getProfileInfoTask.execute(params)
+        val displayName = data[ProfileService.DISPLAY_NAME_KEY] as? String
+        return Optional.from(displayName)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/DefaultProfileService.kt:60]: Can the taskExecutor be removed from the constructor?

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/DefaultProfileService.kt [hunk 3]
```diff
@@ -154,70 +126,38 @@ internal class DefaultProfileService @Inject constructor(private val taskExecuto
...
     }
 
-    override fun addThreePid(threePid: ThreePid, matrixCallback: MatrixCallback<Unit>): Cancelable {
-        return addThreePidTask
-                .configureWith(AddThreePidTask.Params(threePid)) {
-                    callback = matrixCallback
-                }
-                .executeBy(taskExecutor)
+    override suspend fun addThreePid(threePid: ThreePid) {
+        return addThreePidTask.execute(AddThreePidTask.Params(threePid))
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/DefaultProfileService.kt:130]: `return` could be removed

**Review by bmarty — CHANGES_REQUESTED** [review]
Thanks!
Some minor remarks
Benoit

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/DefaultProfileService.kt (hunk 2 above)
Comment (Dominaezzz) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/DefaultProfileService.kt:60]: Not yet. It's used in `refreshThreePids` below.

Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/DefaultProfileService.kt:60]: OK, thanks

**Review by bmarty — APPROVED** [review]
LGTM, thanks for the update

**Pull request merged**